operator/docs/enhancements/zone_aware_replication.md

```diff
@@ -0,0 +1,460 @@
+    //
+    // +optional
+    // +kubebuilder:validation:Optional
+    // +kubebuilder:validation:Minimum:=1
+    // +operator-sdk:csv:customresourcedefinitions:type=spec,xDescriptors="urn:alm:descriptor:com.tectonic.ui:number",displayName="Replication Factor"
+    ReplicationFactor int32 `json:"replicationFactor,omitempty"`
+ 
+    // Replication defines the configuration for Loki data replication
+    //
+    // +optional
+    // +kubebuilder:validation:Optional
+    Replication *ReplicationSpec `json:"replication,omitempty"`
+}
+
+type ReplicationSpec struct {
+    // Factor defines the policy for log stream replication.
+    //
+    // +optional
+    // +kubebuilder:validation:Optional
+    // +kubebuilder:validation:Minimum:=1
+    Factor int32 `json:"factor,omitempty"`
+
+    // Zone is the key that defines a topology in the Nodes' labels
+    //
+    // +required
+    // +kubebuilder:validation:Required
+    Zones []ZoneSpec
+}
+
+// ZoneSpec defines the spec to support zone aware component deployments.
+type ZoneSpec struct {
+    // MaxSkew describes the maximum degree to which Pods can be unevenly distributed
+    //
+    // +optional
+    // +kubebuilder:validation:optional
+    // +kubebuilder:default:="1"
+    MaxSkew int `json:"maxSkew,omitempty"`
+
+    // Topologykey is the key that defines a topology in the Nodes' labels
+    //
+    // +required
+    // +kubebuilder:validation:Required
+    Topologykey string `json:"topologyKey,omitempty"`
+}
+
+```
+
+### Implementation Details/Notes/Constraints
+
+The following manifest represents a full example of a LokiStack with zone-aware data replication turned on using the `topology.kubernetes.io/zone` node label as a key to spread pods across zones and a replication factor of three
+
+```yaml
+apiVersion: loki.grafana.com/v1beta1
+kind: LokiStack
+metadata:
+  name: lokistack-dev
+spec:
+  size: 1x.small
+  storage:
+    secret:
+      name: test
+      type: s3
+  storageClassName: gp2
+  replication:
+    factor: 3
+    zones:
+    - topologyKey: topology.kubernetes.io/zone
+      maxSkew: 1
+```
+
+#### General constraints
+
+### Risks and Mitigations
+
+## Design Details
+
+### 1. Spreading the pods across failure domains using the Kubernetes scheduler (using the PodTopologySpreadConstraint)
+
+The loki components can be divided into the Write Path (Distributor, Ingester) and the Read Path (Query frontend, Querier, Ingester). In the first pass of the proposal do not distinguish between the two paths. See the `Drawbacks` section for more details on this.
+In this step, all componemts and their replicas are configured to spread over different zones. This can be done by making use of the `PodTopologySpreadConstraint` feature in Kubernetes. (See <https://kubernetes.io/docs/concepts/scheduling-eviction/topology-spread-constraints/>)
```

Comment:
```suggestion
In this step, all components and their replicas are configured to spread over different zones. This can be done by making use of the [`PodTopologySpreadConstraint` feature in Kubernetes](https://kubernetes.io/docs/concepts/scheduling-eviction/topology-spread-constraints/).
```